e2e-tests/development-runtime/plugins/gatsby-source-fake-data/api.js

```
@@ -0,0 +1,103 @@
+const fs = require(`fs`)
+const util = require(`util`)
+const path = require(`path`)
+
+const NODE_TYPE = `FakeData`
+
+const readFile = file =>
+  util
+    .promisify(fs.readFile)(file, `utf8`)
+    .then(contents => JSON.parse(contents))
+    .catch(() => [])
+
+const writeFile = (file, contents) =>
+  util
+    .promisify(fs.writeFile)(file, JSON.stringify(contents), `utf8`)
+    .catch(() => null)
```

Nit: Log and then swallow error 😛